modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -346,9 +407,16 @@ private void handleLocalPartitionStatesRequest(LocalPartitionStatesRequest reque
                 if (raftNodeId.groupId() instanceof TablePartitionId) {
                     var tablePartitionId = (TablePartitionId) raftNodeId.groupId();
 
+                    if (!request.partitionIds().isEmpty() && !request.partitionIds().contains(tablePartitionId.partitionId())) {
+                        return;
+                    }
+
                     CatalogTableDescriptor tableDescriptor = catalogManager.table(tablePartitionId.tableId(), catalogVersion);
                     // Only tables that belong to a specific catalog version will be returned.
-                    if (tableDescriptor == null || request.zoneId() != NO_ZONE_ID && tableDescriptor.zoneId() != request.zoneId()) {
+                    if (tableDescriptor == null
```

Comment:
I am a little confused by such a long condition, if possible you can separate it with additional parentheses, for example: `tableDescriptor == null || (!request.zoneIds().isEmpty() && !request.zoneIds().contains(tableDescriptor.zoneId()))`.

I would create a static method that would accept a collection and check the value like this: `!collection.isEmpty() && !collection.contains(e)`.

Reply:
Created a method containsOrEmpty, use it here and for nodes